src/configurationProvider.ts

```diff
@@ -27,10 +44,19 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
         }];
     }
 
-    // Try to add all missing attributes to the debug configuration being launched.
-    public resolveDebugConfiguration(folder: vscode.WorkspaceFolder | undefined, config: vscode.DebugConfiguration, token?: vscode.CancellationToken):
-        vscode.ProviderResult<vscode.DebugConfiguration> {
-        return this.heuristicallyResolveDebugConfiguration(folder, config);
+    private constructLaunchConfigName(mainClass: string, projectName: string, cache: {}) {
+        const prefix = "Debug (Launch)-";
+        let name = prefix + mainClass.substr(mainClass.lastIndexOf(".") + 1);

```

Comment:
no need to add a prefix `Debug (Launch)-`. show the mainClass... directly.

Reply:
we need to tell from attach

Reply:
You have taken the main class as name, apparently it's launch mode. Refer to how eclipse generate debug configuration, main class info is enough because it tells user it's in launch mode implicitly.